charts/consul/templates/crd-grpcroutes.yaml

```diff
@@ -1,769 +1,616 @@
+                                  type: object
+                                type: array
+                            type: object
+                          urlRewrite:
+                            description: URLRewrite defines a schema for a filter
+                              that modifies a request during forwarding.
+                            properties:
+                              pathPrefix:
+                                type: string
+                            type: object
+                        type: object
+                      type: array
+                    matches:
+                      items:
+                        properties:
+                          headers:
+                            description: Headers specifies gRPC request header matchers.
+                              Multiple match values are ANDed together, meaning, a
+                              request MUST match all the specified headers to select
+                              the route.
+                            items:
+                              properties:
+                                name:
                                   type: string
                                 type:
-                                  default: Exact
-                                  description: "Type specifies how to match against the service and/or method. Support: Core (Exact with service and method specified) \n Support: Implementation-specific (Exact with method specified but no service specified) \n Support: Implementation-specific (RegularExpression)"
+                                  description: "HeaderMatchType specifies the semantics
+                                    of how HTTP header values should be compared.
+                                    Valid HeaderMatchType values, along with their
+                                    conformance levels, are: \n Note that values may
+                                    be added to this enum, implementations must ensure
+                                    that unknown values will not cause a crash. \n
+                                    Unknown values here must result in the implementation
+                                    setting the Accepted Condition for the Route to
+                                    status: False, with a Reason of UnsupportedValue."
                                   enum:
-                                    - Exact
-                                    - RegularExpression
+                                  - HEADER_MATCH_TYPE_UNSPECIFIED
+                                  - HEADER_MATCH_TYPE_EXACT
+                                  - HEADER_MATCH_TYPE_REGEX
+                                  - HEADER_MATCH_TYPE_PRESENT
+                                  - HEADER_MATCH_TYPE_PREFIX
+                                  - HEADER_MATCH_TYPE_SUFFIX
+                                  format: int32
+                                  type: string
```

Comment:
From sync conversation: Sounds like this is as expected, and the deserialized format (which _is_ an int, because proto) corresponds to the `format` specifier. The string form is validated prior.